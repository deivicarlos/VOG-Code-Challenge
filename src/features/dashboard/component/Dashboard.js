import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Navbar from '../../../components/Navbar/Navbar';
import Jumbotron from '../../../components/Jumbotron/Jumbotron';

class Dashboard extends PureComponent {
  componentDidMount() {
    const { fetchUserInfo } = this.props;
    fetchUserInfo();
  }
  render() {
    const { logoutUser, user } = this.props;

    return (
      <React.Fragment>
        <Navbar user={user} logoutUser={logoutUser} />
        <Jumbotron />
      </React.Fragment>
    );
  }
}

Dashboard.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  user: PropTypes.object.isRequired,
  fetchUserInfo: PropTypes.func.isRequired,
  logoutUser: PropTypes.func.isRequired,
};

export default Dashboard;
