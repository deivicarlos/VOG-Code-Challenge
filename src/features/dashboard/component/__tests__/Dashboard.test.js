import React from 'react';
import { mount } from 'enzyme';
import Dashboard from '../Dashboard';

describe('Dashboard', () => {
  const logoutUser = jest.fn();
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      isAuthenticated: true,
      user: { firstName: 'Carlos', lastName: 'Garcia' },
      fetchUserInfo: jest.fn(),
      logoutUser,
    };
    wrapper = mount(<Dashboard {...props} />);
  });

  it('it should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call logoutUser on button click', () => {
    wrapper.find('button').prop('onClick')();
    expect(logoutUser).toHaveBeenCalled();
  });
});
