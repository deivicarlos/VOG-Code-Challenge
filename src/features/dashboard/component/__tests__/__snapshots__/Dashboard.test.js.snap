// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard it should render correctly 1`] = `
<Dashboard
  fetchUserInfo={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  isAuthenticated={true}
  logoutUser={[MockFunction]}
  user={
    Object {
      "firstName": "Carlos",
      "lastName": "Garcia",
    }
  }
>
  <Navbar
    logoutUser={[MockFunction]}
    user={
      Object {
        "firstName": "Carlos",
        "lastName": "Garcia",
      }
    }
  >
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark fixed-top"
    >
      <div
        className="d-flex w-100 justify-content-between"
      >
        <div
          className="navbar-brand"
        >
          Carlos Garcia
        </div>
        <div
          className="d-flex align-items-center"
        >
          <button
            className="btn__color__orange px-4"
            onClick={[MockFunction]}
            type="submit"
          >
            Logout
          </button>
        </div>
      </div>
    </nav>
  </Navbar>
  <Jumbotron>
    <div
      className="jumbotron"
    >
      <div
        className="container"
      >
        <h1
          className="display-3 text-center"
        >
          Challenge completed!
        </h1>
        <p
          className="text-center"
        >
          Thank you for the challenge and the opportunity to be evaluated. 
          I hope you like my work and I'm really looking forward to being part of VOG Development team.
        </p>
      </div>
    </div>
  </Jumbotron>
</Dashboard>
`;
